File: testkit/src/main/resources/scenarios/java/Birthday.java
package scenarios.java;

import com.qualityminds.lazyval.LazyValue;
import java.time.LocalDate;

@LazyValue
public record Birthday(LocalDate value) {
    public Birthday {
        if (value == null) {
            throw new IllegalArgumentException("Birthday must not be null");
        }
        if(value.isAfter(LocalDate.now())){
            throw new IllegalArgumentException("Birthday must not be in the future");
        }
    }
}

File: testkit/src/main/resources/scenarios/kotlin/Birthday.kt
package scenarios.kotlin

import com.qualityminds.lazyval.LazyValue;
import java.time.LocalDate;

@LazyValue
data class Birthday(val date: LocalDate){

    init {
        require(!date.isAfter(LocalDate.now())) { "Birthday date cannot be in the future" }
    }
}